### PR TITLE
Add helmclient on controllercontext

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/6639

This is preparing step we move `chartoperator` into app-operator. 
We need helmclient instance to setup chart-operator at TC.

### CODE REVIEW

File: service/controller/app/v1/resource/clients/resource.go [hunk 1]
```diff
@@ -66,6 +78,10 @@ func (r *Resource) addClientsToContext(ctx context.Context, cr v1alpha1.App) err
 		return nil
 	}
 
+	if cc.G8sClient != nil && cc.K8sClient != nil {
```
Comment (tomahawk28) [inline service/controller/app/v1/resource/clients/resource.go]: Return early rather than propagate two indentations blocks.

Comment (xh3b4sd) [reply service/controller/app/v1/resource/clients/resource.go]: Why should the clients be set at this point here? Is the resource called multiple times within the same reconciliation loop? That should not be the case and whenever the resource is called it should mutate the context IMO. Anything else would be confusing and complex.

File: service/controller/app/v1/resource/clients/create.go [hunk 2]
```diff
@@ -21,5 +21,10 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
+	err = r.addHelmClientToContext(ctx, cr)
```
Comment (xh3b4sd) [inline service/controller/app/v1/resource/clients/create.go]: Why don't we add the client in the method that already exists? There are clients added to the context already.

File: service/controller/app/v1/resource/clients/resource.go (hunk 1 above)
Comment (tomahawk28) [reply service/controller/app/v1/resource/clients/resource.go]: That's right, I should not check `nil` at this point, Thanks to pointing out.

File: service/controller/app/v1/resource/clients/create.go (hunk 2 above)
Comment (tomahawk28) [reply service/controller/app/v1/resource/clients/create.go]: Not sure which clients in the method you are meaning, you mean `helmClient`?

Comment (xh3b4sd) [reply service/controller/app/v1/resource/clients/create.go]: There is `addClientsToContext` already. It adds the `g8sClient` and the `k8sClient` to the context. Why do we need another method to add the helm client to the context? Why can't we add the helm client to the context in `addClientsToContext`?

Comment (tomahawk28) [reply service/controller/app/v1/resource/clients/create.go]: I thought I need to differentiate two clients and helm client, but seems that's not the good practice, I'm merging two functions as you said, Thanks a lot!

Comment (tomahawk28) [reply service/controller/app/v1/resource/clients/create.go]: @xh3b4sd PTAL :)

File: service/controller/app/v1/resource/clients/resource.go [hunk 3]
```diff
@@ -84,21 +97,31 @@ func (r *Resource) addClientsToContext(ctx context.Context, cr v1alpha1.App) err
...
-		g8sClient, err := versioned.NewForConfig(restConfig)
-		if err != nil {
-			return microerror.Mask(err)
-		}
-		cc.G8sClient = g8sClient
+	g8sClient, err := versioned.NewForConfig(restConfig)
+	if err != nil {
+		return microerror.Mask(err)
 	}
+	cc.G8sClient = g8sClient
```
Comment (xh3b4sd) [inline service/controller/app/v1/resource/clients/resource.go]: When we could now have this code within a scope block I would be so happy. 🙈

**Pull request merged**